hs_explore/management/commands/lda_recommend_resources.py

```diff
@@ -0,0 +1,342 @@
+        all_subjects.update(subjects)
+        resource_ids.add(res.short_id)
+        resource_to_subjects[res.short_id] = set(subjects)
+    all_subjects_list = list(all_subjects)
+    return resource_to_subjects, all_subjects_list
+
+
+def get_resource_to_abstract():
+    resource_ids = set()
+    resource_to_abstract = {}
+    for res in BaseResource.objects.all():
+        if res is None or res.metadata is None or res.metadata.description is None or \
+           res.metadata.description.abstract is None:
+            continue
+        abstract = res.metadata.description.abstract.lstrip()
+
+        resource_ids.add(res.short_id)
+        resource_to_abstract[res.short_id] = abstract
+
+    return resource_to_abstract
+
+
+def get_resource_to_published():
+    resource_to_published = {}
+    for res in BaseResource.objects.all():
+        if res.raccess.published or res.raccess.public or res.raccess.discoverable:
+            resource_to_published[res.short_id] = True
+        else:
+            resource_to_published[res.short_id] = False
+    return resource_to_published
+
+
+def get_users_interacted_resources(beginning, today):
+    all_usernames = set()
+    user_to_resources_set = defaultdict(set)
+    triples = Variable.user_resource_matrix(beginning, today)
+    for v in triples:
+        username = v[0]
+        res = v[1]
+        all_usernames.add(username)
+        user_to_resources_set[username].add(res)
+    user_to_resources = defaultdict(list)
+    for username, res_ids_set in user_to_resources_set.items():
+        res_ids_list = list(res_ids_set)
+        user_to_resources[username] = list(res_ids_set)
+        user = user_from_id(username)
+        res_list = []
+        for res_id in res_ids_list:
+            try:
+                r = get_resource_by_shortkey(res_id)
+                res_list.append(r)
+            except:
+                continue
+        UserInteractedResources.interact(user, res_list)
+    return user_to_resources, all_usernames
+
+
+def filter_go_words(res_id, doc, resource_to_subjects, go_words, stop):
+    exclude = set(string.punctuation)
+    exclude.remove('_')
+    lemmatizer = WordNetLemmatizer()
+    doc_words = set()
+    doc_list = set()
+
+    for w in doc.split(" "):
+        doc_list.add(lemmatizer.lemmatize(w))
+
+    for go_word in go_words:
+        if " " in go_word:
+            if go_word in doc and go_word not in stop:
+                bigram_name = go_word.replace(" ", "_")
+                doc_words.add(bigram_name)
+        else:
+            if go_word in doc_list and go_word not in stop:
+                doc_words.add(go_word)
+
+    if res_id in resource_to_subjects:
+        res_subjects = resource_to_subjects[res_id]
+        for sub in res_subjects:
+            if sub not in stop:
+                doc_words.add(sub)
+
+    return doc_words
+
+
+def get_resource_to_go_words(resource_to_subjects, resource_to_abstract):
```

Comment:
How much duplication is there in these routines? I just read a set of preparation routines that seem to do the same thing. Eliminate unused code.